src/test/scala/com/redislabs/provider/redis/rdd/BinaryRedisDataframeSuite.scala

```diff
@@ -0,0 +1,48 @@
+package com.redislabs.provider.redis.rdd
+
+import com.redislabs.provider.redis.rdd.Person._
+import org.apache.spark.SparkException
+import org.apache.spark.sql.redis._
+import org.scalatest.Matchers
+
+/**
+  * @author The Viet Nguyen
+  */
+class BinaryRedisDataframeSuite extends RedisStandaloneSuite with Matchers {
+
+  import TestSqlImplicits._
+
+  test("save and load dataframe with binary mode") {
+    val tableName = generateTableName(TableNamePrefix)
+    val df = spark.createDataFrame(data)
+    df.write.format(RedisFormat).option(SqlOptionModel, SqlOptionModelBinary).save(tableName)
+    val loadedDf = spark.read.format(RedisFormat).option(SqlOptionModel, SqlOptionModelBinary)
+      .load(tableName).cache()
+    loadedDf.show()
+    loadedDf.count() shouldBe df.count()
+    loadedDf.schema shouldBe df.schema
+    val loadedArr = loadedDf.as[Person].collect()
+    loadedArr.sortBy(_.name) shouldBe data.toArray.sortBy(_.name)
+  }
+
+  test("save with binary mode and load dataframe") {
+    val tableName = generateTableName(TableNamePrefix)
+    val df = spark.createDataFrame(data)
+    df.write.format(RedisFormat).option(SqlOptionModel, SqlOptionModelBinary).save(tableName)
+    // TODO: support cross mode read/write
```

Comment:
Viet, we don't need to support cross mode read/write. It's a responsibility of user to read with the same model as persisted. It should just fail in runtime. If we can raise a meaningful exception that would be helpful.

Reply:
Sounds like a good option.